sktime/forecasting/prophet.py

```diff
@@ -0,0 +1,341 @@
+        then n_changepoints potential changepoints are selected uniformly from
+        the first `changepoint_range` proportion of the history.
+    changepoint_range: Proportion of history in which trend changepoints will
+        be estimated. Defaults to 0.8 for the first 80%. Not used if
+        `changepoints` is specified.
+    yearly_seasonality: Fit yearly seasonality.
+        Can be 'auto', True, False, or a number of Fourier terms to generate.
+    weekly_seasonality: Fit weekly seasonality.
+        Can be 'auto', True, False, or a number of Fourier terms to generate.
+    daily_seasonality: Fit daily seasonality.
+        Can be 'auto', True, False, or a number of Fourier terms to generate.
+    holidays: pd.DataFrame with columns holiday (string) and ds (date type)
+        and optionally columns lower_window and upper_window which specify a
+        range of days around the date to be included as holidays.
+        lower_window=-2 will include 2 days prior to the date as holidays. Also
+        optionally can have a column prior_scale specifying the prior scale for
+        that holiday.
+    seasonality_mode: 'additive' (default) or 'multiplicative'.
+    seasonality_prior_scale: Parameter modulating the strength of the
+        seasonality model. Larger values allow the model to fit larger seasonal
+        fluctuations, smaller values dampen the seasonality. Can be specified
+        for individual seasonalities using add_seasonality.
+    holidays_prior_scale: Parameter modulating the strength of the holiday
+        components model, unless overridden in the holidays input.
+    changepoint_prior_scale: Parameter modulating the flexibility of the
+        automatic changepoint selection. Large values will allow many
+        changepoints, small values will allow few changepoints.
+    mcmc_samples: Integer, if greater than 0, will do full Bayesian inference
+        with the specified number of MCMC samples. If 0, will do MAP
+        estimation.
+    interval_width: Float, width of the uncertainty intervals provided
+        for the forecast. If mcmc_samples=0, this will be only the uncertainty
+        in the trend using the MAP estimate of the extrapolated generative
+        model. If mcmc.samples>0, this will be integrated over all model
+        parameters, which will include uncertainty in seasonality.
+    uncertainty_samples: Number of simulated draws used to estimate
+        uncertainty intervals. Settings this value to 0 or False will disable
+        uncertainty estimation and speed up the calculation.
+    stan_backend: str as defined in StanBackendEnum default: None - will try to
+        iterate over all available backends and find the working one
+
+    References
+    ----------
+    https://facebook.github.io/prophet
+    https://github.com/facebook/prophet
+
+    """
+
+    def __init__(
+        self,
+        # Args due to wrapping
+        freq,
```

Comment:
`freq` should have a sensible default argument. If `freq` refers to the seasonal periodicity, it should be renamed to `sp` with a default value of `1` (no seasonality) - at least that's what we use elsewhere.

Reply:
freq is related to the DatetimeIndex frequency and has to be a string, see here:
        https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#timeseries-offset-aliases
This link is also there in the Docstring. I was also thinking about setting it to "D" (daily) as default, however I think it is better not to set a default value because the user will not keep in mind to set it. `freq` is needed to create the future DatetimeIndex in case no future index was given. `fbprophet` can not make predictions without a DatetimeIndex. If we set it to "D" and the user actually has a DatetimeIndex with hourly data, the predictions will be on daily basis if he forgets to set `freq`...

Reply:
I see, in that case I'd set it to None and infer it from the given data and if it cannot be inferred, raise an error

Reply:
I believe statsmodels have a function to go between "sp" values and pandas "freq" values, here in case that's helpful: https://github.com/jseabold/statsmodels/blob/master/statsmodels/tsa/tsatools.py#L780

So how is seasonal periodicity set for prophet? Would be nice to have a unified way of doing that for all forecasters (so far we've used the sp keyword) - but probably best to separate that into its own issue/PR.

Reply:
yes thats actually a nice idea. As we enforce DatetimeIndex, the will also be a index.freq value.